src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -603,27 +604,41 @@ export default class MongooseAdapter {
     return words.join(" ");
   }
 
-  static getIdQueryType(idOrIds) {
-    const mode = typeof idOrIds === "string" ? "findOne" : "find";
-    let idQuery;
+  static getIdQueryType(idOrIds, model) {
+    return Q.Promise((resolve, reject) => {
+      const mode = typeof idOrIds === "string" ? "findOne" : "find";
+      let ids = Array.isArray(idOrIds) ? idOrIds : [ idOrIds ];
+      let idQuery;
+      let tests = [];
 
-    if (typeof idOrIds === "string") {
-      if (!this.idIsValid(idOrIds)) {
-        throw new APIError(404, undefined, "No matching resource found.", "Invalid ID.");
-      }
+      ids = ids.filter(id => Boolean(id));
 
-      idQuery = {_id: idOrIds};
-    }
+      ids.forEach(id => {
+        const testDoc = new model({ _id: id })
+        tests.push(testDoc.validate());
+      });
 
-    else if (Array.isArray(idOrIds)) {
-      if (!idOrIds.every(this.idIsValid)) {
-        throw new APIError(400, undefined, "Invalid ID.");
-      }
+      Q.allSettled(tests).then(results => {
```

Comment:
Right. By an id test resolving to false, I meant not just that validation fails, but also that the reason for failure was an error with the id. Connecting that with my definition of isIdValid from [here](https://github.com/ethanresnick/json-api/pull/113#discussion_r55955256), it might look something like:

```js
// assuming idIsValid builds in the full validation check and the truthy filter.
const idValidationPromises = Q.all(ids.map((id) => this.idIsValid(id, model));

// the `then` code here will run as soon immediately once/if an id is reported invalid,
// rather than waiting for every validate check to finish like `allSettled` would have to.
return idValidationPromises.then(() => { 
  /* return the query */ 
}, () => 
  { /* throw a 4xx */ 
});
```